x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/utils/validate.ts

```diff
@@ -128,3 +130,25 @@ function ruleObjectContainsResponseActions(
 ): rule is Rule<UnifiedQueryRuleParams | EsqlRuleParams | EqlRuleParams | NewTermsRuleParams> {
   return rule != null && 'params' in rule && 'responseActions' in rule?.params;
 }
+
+export const validateNonCustomizableUpdateFields = (
+  ruleUpdate: RuleUpdateProps,
+  existingRule: RuleResponse
+) => {
+  if (!isEqual(ruleUpdate.author, existingRule.author)) {
+    throw new ClientError(`Cannot update "author" field for prebuilt rules`, 400);
+  } else if (ruleUpdate.license !== existingRule.license) {
+    throw new ClientError(`Cannot update "license" field for prebuilt rules`, 400);
+  }
+};
+
+export const validateNonCustomizablePatchFields = (
+  rulePatch: RulePatchProps,
+  existingRule: RuleResponse
+) => {
+  if (rulePatch.author && !isEqual(rulePatch.author, existingRule.author)) {
+    throw new ClientError(`Cannot update "author" field for prebuilt rules`, 400);
+  } else if (rulePatch.license && rulePatch.license !== existingRule.license) {
```

Comment:
Let's also handle the case where the incoming licence value is an empty string:

```suggestion
  } else if (rulePatch.license != null && rulePatch.license !== existingRule.license) {
```

Reply:
Good catch, updated